src/main/java/ua/tc/marketplace/controller/CategoryController.java

```diff
@@ -0,0 +1,58 @@
+package ua.tc.marketplace.controller;
+
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import ua.tc.marketplace.model.dto.category.CreateCategoryDTO;
+import ua.tc.marketplace.service.CategoryService;
+
+/**
+ * CategoryController handles HTTP requests related to CRUD operations for the Category entity.
+ * It provides an API for retrieving, creating, updating, and deleting categories.
+ */
+@RestController
+@RequestMapping("/api/v1/category")
+@RequiredArgsConstructor
+public class CategoryController {
+    private static final Logger logger = LoggerFactory.getLogger(CategoryController.class);
```

Comment:
since we use lombok, we can just use `@Slf4j` on class, it will be the same (`log.info()`  and others will be availiable in this class)